src/items/items.cpp

```diff
@@ -452,568 +455,75 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
-			it.levelDoor = pugi::cast<uint32_t>(valueAttribute.value());
-		} else if (tmpStrValue == "maletransformto" || tmpStrValue == "malesleeper") {
-			uint16_t value = pugi::cast<uint16_t>(valueAttribute.value());
-			it.transformToOnUse[PLAYERSEX_MALE] = value;
-			ItemType& other = getItemType(value);
-			if (other.transformToFree == 0) {
-				other.transformToFree = it.id;
-			}
-
-			if (it.transformToOnUse[PLAYERSEX_FEMALE] == 0) {
-				it.transformToOnUse[PLAYERSEX_FEMALE] = value;
-			}
-		} else if (tmpStrValue == "femaletransformto" || tmpStrValue == "femalesleeper") {
-			uint16_t value = pugi::cast<uint16_t>(valueAttribute.value());
-			it.transformToOnUse[PLAYERSEX_FEMALE] = value;
-
-			ItemType& other = getItemType(value);
-			if (other.transformToFree == 0) {
-				other.transformToFree = it.id;
-			}
-
-			if (it.transformToOnUse[PLAYERSEX_MALE] == 0) {
-				it.transformToOnUse[PLAYERSEX_MALE] = value;
-			}
-		} else if (tmpStrValue == "transformto") {
-			it.transformToFree = pugi::cast<uint16_t>(valueAttribute.value());
-		} else if (tmpStrValue == "destroyto") {
-			it.destroyTo = pugi::cast<uint16_t>(valueAttribute.value());
-		} else if (tmpStrValue == "elementice") {
-			Abilities& abilities = it.getAbilities();
-			abilities.elementDamage = pugi::cast<uint16_t>(valueAttribute.value());
-			abilities.elementType = COMBAT_ICEDAMAGE;
-		} else if (tmpStrValue == "elementearth") {
-			Abilities& abilities = it.getAbilities();
-			abilities.elementDamage = pugi::cast<uint16_t>(valueAttribute.value());
-			abilities.elementType = COMBAT_EARTHDAMAGE;
-		} else if (tmpStrValue == "elementfire") {
-			Abilities& abilities = it.getAbilities();
-			abilities.elementDamage = pugi::cast<uint16_t>(valueAttribute.value());
-			abilities.elementType = COMBAT_FIREDAMAGE;
-		} else if (tmpStrValue == "elementenergy") {
-			Abilities& abilities = it.getAbilities();
-			abilities.elementDamage = pugi::cast<uint16_t>(valueAttribute.value());
-			abilities.elementType = COMBAT_ENERGYDAMAGE;
-		} else if (tmpStrValue == "elementdeath") {
-			Abilities& abilities = it.getAbilities();
-			abilities.elementDamage = pugi::cast<uint16_t>(valueAttribute.value());
-			abilities.elementType = COMBAT_DEATHDAMAGE;
-		} else if (tmpStrValue == "elementholy") {
-			Abilities& abilities = it.getAbilities();
-			abilities.elementDamage = pugi::cast<uint16_t>(valueAttribute.value());
-			abilities.elementType = COMBAT_HOLYDAMAGE;
-		} else if (tmpStrValue == "walkstack") {
-			it.walkStack = valueAttribute.as_bool();
-		} else if (tmpStrValue == "blocking") {
-			it.blockSolid = valueAttribute.as_bool();
-		} else if (tmpStrValue == "allowdistread") {
-			it.allowDistRead = booleanString(valueAttribute.as_string());
+		auto parseAttribute = ItemParseAttributesMap.find(tmpStrValue);
+		if (parseAttribute != ItemParseAttributesMap.end()) {
+			// Parse item attributes
+			ItemParse::parseType(tmpStrValue, valueAttribute, itemType);
```

Comment:
This is considerably slower than the previous implementation, albeit more elegant. Is this acceptable? If not, `ItemParseAttributesMap` could work as a dispatcher table, returning a static method pointer?

Reply:
Ah, looks like this should be a comment to #140